chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/RealWorldTestingUploadJob.java

```diff
@@ -0,0 +1,293 @@
+        if (rwtUpload.getType().equals(RealWorldTestingType.PLANS)) {
+            listing.setRwtPlansCheckDate(rwtUpload.getLastChecked());
+            listing.setRwtPlansUrl(rwtUpload.getUrl());
+        } else if (rwtUpload.getType().equals(RealWorldTestingType.RESULTS)) {
+            listing.setRwtResultsCheckDate(rwtUpload.getLastChecked());
+            listing.setRwtResultsUrl(rwtUpload.getUrl());
+        }
+        ListingUpdateRequest request = new ListingUpdateRequest();
+        request.setAcknowledgeWarnings(true);
+        request.setListing(listing);
+        try {
+            certifiedProductManager.update(request);
+        } catch (ValidationException e) {
+            rwtUpload.getErrors().addAll(e.getErrorMessages());
+        } catch (Exception e) {
+            rwtUpload.getErrors().add(e.getMessage());
+        }
+        return rwtUpload;
+    }
+
+    private boolean hasDataChanged(CertifiedProductSearchDetails listing, RealWorldTestingUpload rwtUpload) {
+        if (rwtUpload.getType().equals(RealWorldTestingType.PLANS)) {
+            return !(Objects.equal(listing.getRwtPlansUrl(), rwtUpload.getUrl())
+                    && Objects.equal(listing.getRwtPlansCheckDate(), rwtUpload.getLastChecked()));
+        } else if (rwtUpload.getType().equals(RealWorldTestingType.RESULTS)) {
+            return !(Objects.equal(listing.getRwtResultsUrl(), rwtUpload.getUrl())
+                    && Objects.equal(listing.getRwtResultsCheckDate(), rwtUpload.getLastChecked()));
+        } else {
+            return false;
+        }
+    }
+
+    @SuppressWarnings("checkstyle:linelength")
+    private Optional<CertifiedProductSearchDetails> getListing(String chplProductNumber) {
+        try {
+            return Optional.ofNullable(certifiedProductDetailsManager.getCertifiedProductDetailsByChplProductNumber(chplProductNumber));
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    private List<String> validateRwtUpload(RealWorldTestingUpload rwtUpload) {
+        List<String> errors = new ArrayList<String>();
+
+        //Makes sure all 4 pieces of data are present
+        if (StringUtils.isEmpty(rwtUpload.getChplProductNumber())) {
+            errors.add(errorMessageUtil.getMessage("realWorldTesting.upload.chplProductNumberInvalid"));
+        }
+        if (rwtUpload.getType() == null) {
+            errors.add(errorMessageUtil.getMessage("realWorldTesting.upload.realWorldTestingTypeInvalid"));
+        }
+        if (rwtUpload.getLastChecked() == null) {
+            errors.add(errorMessageUtil.getMessage("realWorldTesting.upload.lastCheckedDateInvalid"));
+        }
+        if (StringUtils.isEmpty(rwtUpload.getUrl())) {
+            errors.add(errorMessageUtil.getMessage("realWorldTesting.upload.url"));
+        }
+        return errors;
+    }
+
+    private void setSecurityContext(UserDTO user) {
+        JWTAuthenticatedUser splitUser = new JWTAuthenticatedUser();
```

Comment:
This variable looks like a copy/paste thing leftover from the split developer job.